app/src/main/java/co/develhope/chooseyouowncocktail_g2/domain/model/Drink.kt

```diff
@@ -8,5 +8,6 @@ data class Drink(
     val category: String?,
     val ingredients: List<String?>,
     val img: String?,
-    val favourite: Boolean
+    var favourite: Boolean,
+    val sortingPosition: Int
```

Comment:
a cosa serve questo campo?

Reply:
per ritrovare la posizione corretta quando si toglie un drink dai preferiti. verra tolto dal momento che abbiamo devciso di fare il sort alfabetico sulla lista originale.

Reply:
Non abbiamo deciso di fare il sort alfabetico, altrimenti stravolgi anche la lista originale che ti arriva dall'API e non risolveresti comunque il problema.
Piuttosto come ti avevo suggerito (e postato lo snippet) potremmo salvare una lista originale, senza preferiti e fare il sort della lista nell'adapter in base a quella

Reply:
a me sembrava propenso a farlo 
il problema si risolverebbe molto piu semplice secondo me.
il quesito e' che che problemi ci sono a stravolgere la lista orriginale ? altrimenti rimango dell idea che sia una complicatezza eccessiva come soluzione. perche anche pesante. mi pare che vincenzo abbia dato 2 soluzioni in merito a  questo.

Reply:
Ordine alfabetico o no, il problema sta nella variabile immutabile sortingPosition.
E se decidessimo di ordinare i drink, per esempio, per categoria e volerne comunque mostrare i preferiti in alto? dovresti aggiornare per ogni singolo elemento della lista la val sortingPosition, con un replaceAll, anche li sarebbe pesante oltre che richiedere più passaggi.
Poi a mio parere, a prescindere non ha molto senso tenere una variabile in  un oggetto per memorizzarne la sua posizione nella lista, dato che esiste la funzione forEachIndexed che per ogni elemento ti restituisce anche la sua posizione